### PR TITLE
Fix MongoDB compilation on El Capitan

### PR DESCRIPTION
**Pull request by patrickod**

### CODE REVIEW

File: files/brews/mongodb.rb [hunk 1]
```diff
@@ -6,6 +6,22 @@ class Mongodb < Formula
   stable do
     url "https://fastdl.mongodb.org/src/mongodb-src-r2.6.8.tar.gz"
     sha1 "d67254ef3ba5ba81e21c56be9b919c3a10e01b32"
+
+    # Review this patch with the next stable release.
+    # Note it is a different patch to the one applied to all builds further below.
+    # This is already fixed in the devel & HEAD builds.
+    if MacOS.version == :yosemite
+      patch do
+        url "https://github.com/mongodb/mongo/commit/759b6e8.diff"
```
Comment (eugeneius) [inline files/brews/mongodb.rb]: The module installs [MongoDB r2.6.8](https://github.com/intercom/puppet-mongodb/blob/537a85bfb3ecbe2a63b662c96a4b0aa4006472b4/files/brews/mongodb.rb#L7), which already contains this commit (as https://github.com/mongodb/mongo/commit/8ffd49eda26de47acb766b4094f7aba4ddb7a25d).

MongoDB currently compiles successfully through Boxen on Yosemite, I don't think we should mess with it here.

Comment (antidis) [inline files/brews/mongodb.rb]: @patrickod can you follow up here please? We're all anxious :)

Comment (patrickod) [inline files/brews/mongodb.rb]: yep going to take some time today w/ @eugeneius to merge these

Comment (antidis) [inline files/brews/mongodb.rb]: NICE

**Comment by patrickod** [comment]
@eugeneius removed the Yosemite patch. Now diff only includes patch for El Capitan, and extends the compiler flag check for disabling warnings to El Capitan.

**Comment by eugeneius** [comment]
Wonderful :sparkles: :ship:

**Comment by patrickod** [comment]
@eugeneius don't have merge powers. Can you merge?

**Comment by eugeneius** [comment]
Oh, neither do I :unamused: @mstewart?

**Comment by mstewart** [comment]
Whoops, didn't realise I needed to manually add the intercom org as owners.  Added all of engineering.

**Comment by eugeneius** [comment]
Thanks! :ok_hand:

**Pull request merged**